tests/components/hunterdouglas_powerview/conftest.py

```diff
@@ -1,47 +1,161 @@
+@pytest.fixture
+@pytest.mark.usefixtures("mock_hunterdouglas_base")
+async def mock_hunterdouglas_user(
+    device_json: str,
+) -> Generator[MagicMock, None, None]:
+    """Return a mocked Powerview Hub with only base raw data."""
+    with patch(
+        "homeassistant.components.hunterdouglas_powerview.Hub.request_raw_data",
+        return_value=load_json_object_fixture(device_json, DOMAIN),
+    ):
+        yield
 
 
 @pytest.fixture
-def mock_powerview_v2_hub(powerview_userdata, powerview_fwversion, powerview_scenes):
-    """Mock a Powerview v2 hub."""
+@pytest.mark.usefixtures("mock_hunterdouglas_base")
+async def mock_hunterdouglas_full(
+    device_json: str,
+    home_json: str,
+    firmware_json: str,
+    rooms_json: str,
+    scenes_json: str,
+    shades_json: str,
+) -> Generator[MagicMock, None, None]:
+    """Return a mocked Powerview Hub with all data populated."""
     with patch(
-        "homeassistant.components.hunterdouglas_powerview.UserData.get_resources",
-        return_value=powerview_userdata,
+        "homeassistant.components.hunterdouglas_powerview.Hub.request_raw_data",
+        return_value=load_json_object_fixture(device_json, DOMAIN),
+    ), patch(
+        "homeassistant.components.hunterdouglas_powerview.Hub.request_home_data",
+        return_value=load_json_object_fixture(home_json, DOMAIN),
+    ), patch(
+        "homeassistant.components.hunterdouglas_powerview.Hub.request_raw_firmware",
+        return_value=load_json_object_fixture(firmware_json, DOMAIN),
     ), patch(
         "homeassistant.components.hunterdouglas_powerview.Rooms.get_resources",
-        return_value={"roomData": []},
+        return_value=load_json_value_fixture(rooms_json, DOMAIN),
     ), patch(
         "homeassistant.components.hunterdouglas_powerview.Scenes.get_resources",
-        return_value=powerview_scenes,
+        return_value=load_json_value_fixture(scenes_json, DOMAIN),
     ), patch(
         "homeassistant.components.hunterdouglas_powerview.Shades.get_resources",
-        return_value={"shadeData": []},
+        return_value=load_json_value_fixture(shades_json, DOMAIN),
     ), patch(
-        "homeassistant.components.hunterdouglas_powerview.ApiEntryPoint",
-        return_value=powerview_fwversion,
+        "homeassistant.components.hunterdouglas_powerview.cover.BaseShade.refresh",
+    ), patch(
+        "homeassistant.components.hunterdouglas_powerview.cover.BaseShade.current_position",
+        new_callable=PropertyMock,
+        return_value=ShadePosition(primary=0, secondary=0, tilt=0, velocity=0),
     ):
         yield
+
+
+@pytest.fixture
+def device_json(api_version: int) -> str:
+    """Return the request_raw_data fixture for a specific device."""
+    if api_version == 1:
+        return "gen1/userdata.json"
+    if api_version == 2:
+        return "gen2/userdata.json"
+    if api_version == 3:
+        return "gen3/gateway/primary.json"
```

Comment:
If we ever have more it would be better to write these as constant dict lookups to reduce time complexity

https://wiki.python.org/moin/TimeComplexity

fixture = DEVICE_JSON_FIXTURE[api_version]